common/djangoapps/request_cache/middleware.py

```diff
@@ -108,7 +109,7 @@ def func_call_cache_key(func, *args, **kwargs):
     the function's name, and a stringified list of arguments
     and a query string-style stringified list of keyword arguments.
     """
-    converted_args = map(str, args)
-    converted_kwargs = map(str, reduce(list.__add__, map(list, sorted(kwargs.iteritems())), []))
+    converted_args = map(force_text, args)
+    converted_kwargs = map(force_text, reduce(list.__add__, map(list, sorted(kwargs.iteritems())), []))
```

Comment:
Hi @muzaffaryousaf , I will be answering questions here until OmarIthawi is able to attend to them.

If any of the characters inside `args` or `kwargs` is non-ASCII, then `UnicodeEncodeError` will be raised if str is used, force_text will be able to handle such exceptions better, and that is the whole point of the PR, to handle Unicode comments